reV/SAM/generation.py

```diff
@@ -1149,6 +1161,221 @@ def default():
         return DefaultTroughPhysicalProcessHeat.default()
 
 
+class Geothermal(AbstractSamGenerationFromWeatherFile):
+    """Class for geothermal generation from SAM"""
+
+    MODULE = 'geothermal'
+    PYSAM = PySamGeothermal
+    PYSAM_WEATHER_TAG = "file_name"
+
+    @staticmethod
+    def default():
+        """Get the executed default PySAM Geothermal object.
+
+        Returns
+        -------
+        PySAM.Geothermal
+        """
+        return DefaultGeothermal.default()
+
+    def cf_profile(self):
+        """Get hourly capacity factor (frac) profile in local timezone.
+        See self.outputs attribute for collected output data in UTC.
+
+        Returns
+        -------
+        cf_profile : np.ndarray
+            1D numpy array of capacity factor profile.
+            Datatype is float32 and array length is 8760*time_interval.
+        """
+        return self.gen_profile() / self.sam_sys_inputs['nameplate']
+
+    def assign_inputs(self):
+        """Assign the self.sam_sys_inputs attribute to the PySAM object."""
+        if self.sam_sys_inputs.get("ui_calculations_only"):
+            msg = ('reV requires model run - cannot set '
+                   '"ui_calculations_only" to `True` (1). Automatically '
+                   'setting to `False` (0)!')
+            logger.warning(msg)
+            warn(msg)
+            self.sam_sys_inputs["ui_calculations_only"] = 0
+        super().assign_inputs()
+
+    def set_resource_data(self, resource, meta):
+        """Generate the weather file and set the path as an input.
+
+        The Geothermal PySAM model requires a data file, not raw data.
```

Comment:
ugh another one of these huh? annoying that this is still a thing. Nice work using the other similar frameworks though.

Reply:
Yea, don't even get me started lol. *They don't even use the weather data*.